### PR TITLE
Use composite actions for gradle pre/post build steps

### PR DESCRIPTION
**Pull request by dradosevic**

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -49,76 +49,45 @@ jobs:
...
+          java: ${{ matrix.java }}
+          gradle-enterprise-access-key: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
+          gradle-enterprise-cache-username: ${{ secrets.GRADLE_ENTERPRISE_CACHE_USERNAME }}
+          gradle-enterprise-cache-password: ${{ secrets.GRADLE_ENTERPRISE_CACHE_PASSWORD }}
       - name: Optional tear down step
         env:
           K8S_VERSION: ${{ matrix.k8s }}
           JAVA_VERSION: ${{ matrix.java }}
         run: |
           [ -f ./teardown.sh ] && ./teardown.sh || true
```
Comment (alvarosanchez) [inline .github/workflows/gradle.yml:79]: Perhaps this step could be moved to the `post-build` action

**Review by alvarosanchez — APPROVED** [review]

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (dradosevic) [reply .github/workflows/gradle.yml]: Moved optional tear down step to the `post-build` action

File: .github/actions/gradle-build/action.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    required: true
+    type: string
+runs:
+  using: "composite"
+  steps:
+    - name: Build with Gradle
+      shell: bash
+      run: |
+        # Awful hack for kapt and JDK 16. See https://youtrack.jetbrains.com/issue/KT-45545
+        if [ ${{ inputs.java }} == 16 ]; then export GRADLE_OPTS="-Dorg.gradle.jvmargs=--illegal-access=permit"; fi
```
Comment (alvarosanchez) [inline .github/actions/gradle-build/action.yml:28]: This can probably be removed now, the linked bug is fixed

Comment (dradosevic) [reply .github/actions/gradle-build/action.yml]: removed this fix

**Pull request merged**